### PR TITLE
Useorm

### PR DESCRIPTION
**Pull request by TordJoranger**

### CODE REVIEW

File: Core/Misc/TopicHelper.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+public static class TopicHelper
+{
+    public static IEnumerable<string> GetAllTopicsAsEnumerable()
+    {
+        return typeof(PcsTopicConstants)
+            .GetFields(BindingFlags.Public | BindingFlags.Static | BindingFlags.FlattenHierarchy)
+            .Where(fi => fi is { IsLiteral: true, IsInitOnly: false })
+            .Select(x => x.GetRawConstantValue()?.ToString())
+            .Where(t => t is not PcsTopicConstants.WorkOrderCutoff and not PcsTopicConstants.Document and not PcsTopicConstants.Milestone)
```
Comment (jsoi-eq) [inline Core/Misc/TopicHelper.cs:14]: This filter kind of invalidates the method name GetAllxxx? Could lead to confusion?

**Comment by jsoi-eq** [comment]
Missing reference to DevOps workitem in PR Description.

File: Core/Services/FamFeederService.cs [hunk 2]
```diff
@@ -82,9 +81,9 @@ public async Task<string> RunForCutoffWeek(string cutoffWeek, string plant, ILog
         }
 
         _logger.LogInformation(
-            "Found {events} events for WoCutoff for week {cutoffWeek} and plant {plant} ", events.Count, cutoffWeek, plant);
+            "Found {Events} events for WoCutoff for week {CutoffWeek} and plant {Plant} ", events.Count, cutoffWeek, plant);
```
Comment (jsoi-eq) [inline Core/Services/FamFeederService.cs:84]: Same

File: Core/Services/FamFeederService.cs [hunk 3]
```diff
@@ -156,105 +155,105 @@ private async Task<List<string>> GetEventsBasedOnTopicAndPlant(QueryParameters q
         var plant = queryParameters.Plant;
         switch (queryParameters.PcsTopic)
```
Comment (jsoi-eq) [inline Core/Services/FamFeederService.cs:156]: Could you use pattern matching for more compact syntax?

        events = queryParameters.PcsTopic switch
        {
            PcsTopicConstants.SwcrAttachment => await _repo.GetSwcrAttachments(plant),
            nameof(PcsTopicConstants.SwcrType) => await _repo.GetSwcrType(plant)
        };

File: FamFeederFunction/Functions/SearchFeeder/SearchFeederFunction.cs [hunk 4]
```diff
@@ -39,11 +38,11 @@ public SearchFeederFunction(ISearchFeederService searchFeederService)
             return new BadRequestObjectResult("Please provide both plant and topic");
         }
 
-        var parsed = TryParse(topicString, out PcsTopic _);
-        if (!parsed)
-        {
-            return new BadRequestObjectResult("Please provide valid topic");
-        }
+        // var parsed = TryParse(topicString, out PcsTopic _);
```
Comment (jsoi-eq) [inline FamFeederFunction/Functions/SearchFeeder/SearchFeederFunction.cs:41]: Remove commented out code.

File: Infrastructure/Handlers/GuidTypeHandler.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+{
+    public static readonly GuidTypeHandler Default = new();
+
+    public override Guid Parse(object value)
+    {
+        var result = Guid.TryParse(value.ToString(), out var guid)
+            ? guid
+            : Guid.TryParse(ByteArrayToHexString((byte[])value), out var guid2)
+                ? guid2
+                :throw new Exception("Could not parse Guid");
```
Comment (jsoi-eq) [inline Infrastructure/Handlers/GuidTypeHandler.cs:17]: Can we return value.ToString() as info in the exception message or does it not make sense?

**Review by jsoi-eq — COMMENTED** [review]
Waiting for feedback

File: Infrastructure/Handlers/BooleanHandler.cs [hunk 6]
```diff
@@ -0,0 +1,13 @@
+﻿using System.Data;
+using Dapper;
+
+namespace Infrastructure.Handlers;
+
+internal class BooleanHandler : SqlMapper.TypeHandler<bool>
+{
+    public static readonly BooleanHandler Default = new();
+
+    public override bool Parse(object value) => value.ToString() == "Y";
```
[the remark this answers is not in the document]
Comment (TordJoranger) [reply Infrastructure/Handlers/BooleanHandler.cs:10]: I think not, but I've renamed to name to include Oracle

**Pull request merged**